src/sentry/static/sentry/app/components/shareIssue.tsx

```diff
@@ -0,0 +1,242 @@
+
+  handleToggleShare = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    this.props.onToggle();
+  };
+
+  handleOpen = () => {
+    const {loading, isShared, onToggle} = this.props;
+    if (!loading && !isShared) {
+      // Starts sharing as soon as dropdown is opened
+      onToggle();
+    }
+  };
+
+  // State of confirm modal so we can keep dropdown menu opn
+  handleConfirmCancel = () => (this.hasConfirmModal = false);
+  handleConfirmReshare = () => (this.hasConfirmModal = true);
+
+  render() {
+    const {loading, isShared, shareUrl, onReshare} = this.props;
+
+    return (
+      <DropdownLink
+        className="share-issue btn-sm btn btn-default"
+        shouldIgnoreClickOutside={() => this.hasConfirmModal}
+        title={
+          <DropdownTitleContent>
+            <IndicatorDot isShared={isShared} />
+            {t('Share')}
+          </DropdownTitleContent>
+        }
+        onOpen={this.handleOpen}
+        keepMenuOpen
+      >
+        <DropdownContent>
+          <Header>
+            <Title>{t('Enable public share link')}</Title>
+            <Switch isActive={isShared} size="sm" toggle={this.handleToggleShare} />
+          </Header>
+
+          {loading && (
+            <LoadingContainer>
+              <LoadingIndicator mini />
+            </LoadingContainer>
+          )}
+
+          {!loading && isShared && shareUrl && (
+            <ShareUrlContainer
+              shareUrl={shareUrl}
+              onCancel={this.handleConfirmCancel}
+              onConfirming={this.handleConfirmReshare}
+              onConfirm={onReshare}
+            />
+          )}
+        </DropdownContent>
+      </DropdownLink>
+    );
+  }
+}
+
+export default ShareIssue;
+
+const UrlContainer = styled('div')`
+  display: flex;
+  align-items: stretch;
+  border: 1px solid ${p => p.theme.borderDark};
+  border-radius: ${space(0.5)};
+`;
+
+const LoadingContainer = styled('div')`
+  display: flex;
+  justify-content: center;
+`;
+
+const DropdownTitleContent = styled('div')`
+  display: flex;
+  align-items: center;
+  margin-right: ${space(0.5)};
+`;
+
+const DropdownContent = styled('li')`
+  padding: ${space(1.5)} ${space(2)};
+
+  > div:not(:last-of-type) {
+    margin-bottom: ${space(1.5)};
+  }
+`;
+
+const Header = styled('div')`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const Title = styled('h6')`
+  margin: 0;
+  padding-right: ${space(4)};
+  white-space: nowrap;
+`;
+
+const IndicatorDot = styled('span')<{isShared?: boolean}>`
+  display: inline-block;
+  margin-right: ${space(0.5)};
+  border-radius: 50%;
+  width: 10px;
+  height: 10px;
+  background: ${p => (p.isShared ? p.theme.green400 : p.theme.borderDark)};
+`;
+
+const StyledAutoSelectText = styled(AutoSelectText)<
+  React.ComponentPropsWithRef<typeof AutoSelectText>
+>`
+  flex: 1;
+  padding: ${space(0.5)} 0 ${space(0.5)} ${space(0.75)};
+  ${overflowEllipsis}
+`;
+
+const StyledTextContainer = styled('div')`
```

Comment:
as you are not styling another component here, I'd name it only `TextContainer`  😉